qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -152,10 +157,129 @@ void assertBasicSearchWorks(String index) throws IOException {
+        @SuppressWarnings("unchecked")
+        Map<String, Object> hit = (Map<String, Object>)
+                ((List)(XContentMapValues.extractValue("hits.hits", searchRsp))).get(0);
+        String docId = (String) hit.get("_id");
+
+        requestBody = "{ \"doc\" : { \"foo\": \"bar\"}}";
+        response = client().performRequest("POST", "/" + index + "/doc/" + docId + "/_update", Collections.emptyMap(),
+                new StringEntity(requestBody, ContentType.APPLICATION_JSON));
+        assertEquals(200, response.getStatusLine().getStatusCode());
+
+        Map<String, Object> getRsp = toMap(client().performRequest("GET", "/" + index + "/doc/" + docId));
+        @SuppressWarnings("unchecked")
+        Map<String, Object> source = (Map<String, Object>) getRsp.get("_source");
+        assertTrue("doc does not contain 'foo' key: " + source, source.containsKey("foo"));
+
+        requestBody = "{ \"index\": { \"refresh_interval\" : \"1s\" }}";
+        response = client().performRequest("PUT", "/" + index + "/_settings", Collections.emptyMap(),
+                new StringEntity(requestBody, ContentType.APPLICATION_JSON));
+        assertEquals(200, response.getStatusLine().getStatusCode());
+    }
+
+    void assertNewReplicasWork(String index) throws Exception {
+        final int numReplicas = 1;
+        final long startTime = System.currentTimeMillis();
+        logger.debug("--> creating [{}] replicas for index [{}]", numReplicas, index);
+        String requestBody = "{ \"index\": { \"number_of_replicas\" : " + numReplicas + " }}";
+        Response response = client().performRequest("PUT", "/" + index + "/_settings", Collections.emptyMap(),
+                new StringEntity(requestBody, ContentType.APPLICATION_JSON));
+        assertEquals(200, response.getStatusLine().getStatusCode());
+
+        Map<String, String> params = new HashMap<>();
+        params.put("timeout", "2m");
+        params.put("wait_for_status", "green");
+        params.put("wait_for_no_relocating_shards", "true");
+        params.put("wait_for_events", "languid");
+        Map<String, Object> healthRsp = toMap(client().performRequest("GET", "/_cluster/health/" + index, params));
+        assertEquals("green", healthRsp.get("status"));
+        assertFalse((Boolean) healthRsp.get("timed_out"));
+
+        logger.debug("--> index [{}] is green, took [{}] ms", index, (System.currentTimeMillis() - startTime));
+        Map<String, Object> recoverRsp = toMap(client().performRequest("GET", "/" + index + "/_recovery"));
+        logger.debug("--> recovery status:\n{}", recoverRsp);
+
+        // TODO: do something with the replicas! query? index?
+    }
+
     static Map<String, Object> toMap(Response response) throws IOException {
         return XContentHelper.convertToMap(JsonXContent.jsonXContent, EntityUtils.toString(response.getEntity()), false);
     }
 
+    static String toStr(Response response) throws IOException {
```

Comment:
I do this a bit in other places. Do you want to convert? I didn't feel the need to make this but it *is* shorter so I'm fine with it.